compiler/rustc_span/src/source_map.rs

```diff
@@ -462,6 +462,17 @@ impl SourceMap {
         )
     }
 
+    pub fn span_to_locatio_info(&self, sp: Span) -> (Symbol, usize, usize, usize, usize) {
+        if self.files.borrow().source_files.is_empty() || sp.is_dummy() {
```

Comment:
What is the `self.files.borrow().source_files.is_empty()` condition necessary for?

Reply:
this has not been answered yet.

Reply:
i based it on the following. my guess is it handles the temporary files generated by the compiler.

https://github.com/rust-lang/rust/blob/9aa5c24b7d763fb98d998819571128ff2eb8a3ca/compiler/rustc_span/src/source_map.rs#L441-L448

ps: pretty sure responded to this. not sure why my comments got reset.

Reply:
I think `span_to_string` should call `span_to_location_info` to get the information it needs. This avoids the duplication of all the logic.

Reply:
this is also done.